Dk Diseases +15% (Patch 3.2) / Sanctified Wrath / Scourge Strike / Wandering Plague / Renewed Hope + Flash heal

P-Kito opened the pull request:
…absorb effects

P-Kito committed:
Unit/CalcAbsorbResist: Paladin talent Sanctified Wrath should pierce absorb effects

P-Kito committed:
Dk diseases should make 15% more damage 3.2.0

P-Kito commented:
Wtf how did this thing go in there <_<

P-Kito commented:
Unfortunately it made one PR for 2 things.

P-Kito committed:
Spell/DK: Scourge Strike should deal shadow damage even if the main damage was absorbed

P-Kito commented:
Unfortunately it made one PR for 3 things.

Kittnz commented on src/server/game/Entities/Unit/Unit.cpp:
```diff
@@ -10278,6 +10278,11 @@ float Unit::SpellDamagePctDone(Unit* victim, SpellInfo const* spellProto, Damage
                         AddPct(DoneTotalMod, aurEff->GetAmount());
+
+            // Diseases should make 15% more damage with patch 3.2.0
+            if (spellProto->SpellFamilyFlags[1] & 0x6000000 && spellProto->SpellFamilyFlags[1] & 0x800)
```
no name for 0x6000000 ?

ccrs replied:
there is no definition for spellfamilyflags

P-Kito committed:
Spell/DK: Wandering Plague should not keep players infight

Players can't drink for ages in arena when they have diseases on themselves by a dk with this talent. Every tick keeps them infight.

P-Kito committed:
Unit/Priest: Renewed Hope (Talent) should also affect Flash heal. DBC is missing flash heal in spell overrides.

P-Kito committed:
Chance CalculatePctF to CalculatePct

P-Kito committed:
Change full_crit_chance to crit_chance

Sry I am on an older version of TC :)

Nyeriah commented:
This PR can't be reviewed as it is. Please create a new branch for each PR so it does not include unwanted changes.

Nyeriah commented on src/server/scripts/Spells/spell_dk.cpp:
```diff
@@ -1445,6 +1445,7 @@ class spell_dk_scourge_strike : public SpellScriptLoader
 
         private:
             float multiplier;
+            int32 damageDone;
```
int32 damageDone is left uninitialized.

jackpoz commented on src/server/game/Entities/Unit/Unit.cpp:
```diff
@@ -1647,6 +1647,9 @@ void Unit::CalcAbsorbResist(Unit* victim, SpellSchoolMask schoolMask, DamageEffe
 
     RoundToInterval(auraAbsorbMod, 0.0f, 100.0f);
 
+    auto absorb_ignoring_damage = CalculatePct(dmgInfo.GetDamage(), auraAbsorbMod);
```
please don't use the auto keyword for such simple values as the return type of CalculatePct()

jackpoz commented:
This PR has the purpose of fixing some spells but it modifies code that affects other spells too. Please verify that all other spells are working correctly or at least that no new issues are added.

P-Kito commented:
Which other spells does it effect? The Absorb-stuff is only for sanctified wrath, since that's pretty much the only spell with this spell_aura_mod_.... 

I see no reason to not use the auto keyboard there.

It was enough work to create this PR. It's working for me as it should. Maybe this brings some ideas for the recent contributors to make a PR of your dreams.

tkrokli commented:
I agree. This PR has got 5 different fixes mixed into 1 PR. It should have been 5 PRs, 1 for each commit.

P-Kito commented:
I dont even know how to do this, since every commit I did was just attached to this :/
Well, sorry I was b1tching around on my comment. Probaby someday I will change it... or someone else :)

tkrokli commented:
@P-Kito : it should be simple enough.
If you just use git or GitExtensions to create a new **branch** based on an updated straight TC 3.3.5 source.

Name them something short, like:
- "CalcAbsorbResist"
- "dk_disease_dmg"
- "dk_ss_shadow_dmg"
- "dk_spell_wp_pvp"
- "priest_custom_crit_rh_fh"

Switch to each branch. Apply only 1 of the 5 first commits in this PR to each of those branches. :)

ccrs commented:
@tkrokli it's simple, anyone can do it. Just takes too much time :/

tkrokli commented:
I got the time, if P-Kito does not feel like doing it. It's just that I'm not good at coding.

P-Kito commented:
Go ahead if you like to do it. I appreciate that you take the time.

tkrokli commented:
OK, very well. If needed, I can add collaborator access to my branches. I'll do it as I wrote above.

P-Kito commented:
Thank you VERY VERY MUCH. 
Let me know if I can do anything for you.

tkrokli commented:
Well, there is the issue of hardcoded values in https://github.com/TrinityCore/TrinityCore/pull/15037/files
I was looking for a way to replace
```cpp
if (AuraEffect* aurEff = GetAuraEffectOfRankedSpell(57470, EFFECT_0))
```
with
```cpp
if (AuraEffect* aurEff = GetAuraEffectOfRankedSpell(SPELL_PRIEST_RENEWED_HOPE, EFFECT_0))
```
but that would return spell ID `63944` instead of `57470`. Any idea where to look up a name for `spell 57470` or if it needs to be created in spell_priest.cpp or something like that? (+ similar for `SpellIconID 177`).

ccrs commented:
there are also no definitions for spelliconid

Aokromes commented:
#15033 #15034 #15035 #15036 #15037
